pyoneer/rl/rollouts/unroll_ops.py

```diff
@@ -0,0 +1,146 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+
+import tensorflow as tf
+
+
+class Unroll(collections.namedtuple(
+        'Unroll', ['outputs', 'initializer', 'initial_time_step'])):
+    pass
+
+
+def n_step_unroll(n_step,
+                  initializer,
+                  agent_step_fn,
+                  env_step_fn,
+                  initial_time_step=0):
+    """Unroll interactions for n-steps.
+
+    Args:
+        n_step: Tensor of shape [].
+        initializer: Tuple containing initial values:
+            (agent_output, env_output)
+        agent_step_fn: Agent step function:
+            fn(env_output, agent_output, time_step)
+        env_step_fn: Environment step function:
+            fn(agent_output, env_output, time_step)
+        initial_time_step: The initial step value.
```

Comment:
What's the use case of having an argument for this?

Reply:
This allows you to track meta-data for rollouts seen from the agent/environment point-of-view in a stateful way. So if you wanted to resume a rollout where you left off, you would pass the final time step of the previous rollout as the initial time step of this rollout.